### DISCUSSION TITLE
I can't connect to a remote private repo in vscode on Mac

### DISCUSSION THREAD

**Discussion by jaliubusiness** [community/community#66813] [Repositories] [answered]
### Select Topic Area

Question

### Body

I have two GitHub accounts - one with my college, and one personal for business purposes.

I have a private repo on each, let's refer to them as college repo and personal repo. 

I created both of them on my windows pc. I am able to connect to the college repo on my Mac by going into the vscode console and doing git init, git remote add origin college-repo-url, and then git pull. This works. 

When I do the same on my personal repo, it does not. When I type git pull, it says that the repo is not found, which is very weird. git remote -v shows me that it IS there. But I do not know what the difference is between my personal repo and my college repo that I can pull the college one but not the personal one. These are both private repos. 

Google searches have yielded no results either. I deleted all traces of GitHub from keychain access to no avail.

SOS.

**Comment by seyLu** [comment, marked as answer]
#### Hi @jaliubusiness,

For you to be able to access a private repo that is not yours, you have to at least be a collaborator on that repo. Github sees this two different Github accounts as two different users.

Go to settings > collaborators:
- click add people
- type in and select their github username
- the collaborator should receive a notification that you gave them an invite
- once they accept the invite, then they should be able to access the private repo as normal (of course without the admin privileges, unless you elevate their role/permissions)
![image](https://github.com/community/community/assets/98249191/6f704fe5-a4cc-4817-b372-daab24085380)
![image](https://github.com/community/community/assets/98249191/720864c2-6e07-4dfe-a89b-75154d77dfec)

**Reply by jaliubusiness** [reply]
Hi @seyLu ,

Thank you so much. This worked. 

Do you mind clearing some things up for me? It seems that I am under some false impressions.

To me this implies that my college repo, is my 'main' repo(according to vscode), so to speak, as I could connect to it in my vscode, but since my personal repo isn't my main repo, I have to add my main repo as a collaborator. It seems like my vscode recognizes my college repo but not my personal repo if that makes sense? Could you shed some light?

**Reply by seyLu** [reply]
> To me this implies that my college repo, is my 'main' repo(according to vscode), so to speak, as I could connect to it in my vscode, but since my personal repo isn't my main repo, I have to add my main repo as a collaborator. It seems like my vscode recognizes my college repo but not my personal repo if that makes sense? Could you shed some light?

You're adding users to collaborate on a repo, not the other way around. VSCode doesn't care about the current workstation/folder/proj you're working, it's git that cares. If the current git user does not have access to a repo, it would throw a permission error, or in this case a private repo, not found error.

Github is a cloud platform that hosts git repos. 

If you have github student pack, you can use a tool called GitKraken free for a year, and switching between multiple git credentials is very simple there.

![image](https://github.com/community/community/assets/98249191/e4a4b3c1-0517-4082-9278-46811b68b77e)
